README.md

```diff
@@ -1,2 +1,118 @@
-# juttle-engine
-juttle execution engine
+# Juttle Engine
+
+[![Build Status](https://travis-ci.org/juttle/juttle-engine.svg?branch=master)](https://travis-ci.org/juttle/juttle-engine)
+
+Juttle Engine is an integrated environment for developing and executing
+[juttle](https://github.com/juttle/juttle) programs and visualizations. It lets
+you run Juttle programs stored on the file system and present the results in a
+browser for experimentation, development, deployment, and debugging of juttle
+programs.
+
+When using Juttle to power data visualizations, they cannot be rendered in the
+Juttle CLI. Juttle Engine bridges this gap by including a browser based
+execution application with a javascript charting library that receives streaming
+data from Juttle programs running server-side via a websocket. It also includes
+user-interface logic to implement Juttle input controls.
+
+![](docs/screenshots/stock_prices_example.png)
+
+If you simply want to try running some basic juttle programs that output directly to the terminal, you can use the ``juttle`` command line program included in the Juttle respository.
+
+Under the covers, Juttle Engine integrates
+[juttle-service](https://github.com/juttle/juttle-service), a node.js API server
+that enables execution of Juttle programs using a REST API with the ability to
+serve web application bundles. It comes with the
+[juttle-viewer](https://github.com/juttle/juttle-viewer) application that
```

Comment:
`juttle-client-library` is missing from this list of pieces but that may be a few too many layers to explain. Fine as is.